### PR TITLE
Run CI Python tasks on Fedora container

### PR DESCRIPTION
**Pull request by bgurney-rh**
Related: stratis-storage/project#294

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -19,36 +19,30 @@ jobs:
...
-          - python-version: 3.8.6
-            dependencies: >
-              dbus-python==1.2.16
-              pylint==2.4.4
-              psutil==5.6.7
+          - dependencies: >
+              dbus-python-devel
+              pylint
+              python3-psutil
+            image: fedora:33
```
Comment (mulkieran) [inline .github/workflows/main.yml:26]: Set back to fedora:33, we definitely need to work through w/ the correct values.

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -19,36 +19,30 @@ jobs:
...
             task: PYTHONPATH=./src make -f Makefile lint
-          - python-version: 3.8.6
-            dependencies: black==20.8b1 isort==4.3.21
+          - dependencies: >
+              black
+              python3-isort
+            image: fedora:28
```
Comment (mulkieran) [inline .github/workflows/main.yml:31]: formatting + linting should also be current development fedora release, so this should be 32 also.

**Review by mulkieran — CHANGES_REQUESTED** [review]
Ok! It's maybe just the formatting/linting task that is causing the problem. Try fixing that up, and _probably_ this will pass.

File: .github/workflows/main.yml (hunk 1 above)
Comment (bgurney-rh) [reply .github/workflows/main.yml:26]: It says `fedora:33` right now, in the "make lint" task.  Should it say something else?

Comment (mulkieran) [reply .github/workflows/main.yml:26]: > It says `fedora:33` right now, in the "make lint" task. Should it say something else?

Whoops! I meant "fedora:32".

**Comment by bgurney-rh** [comment]
Now I'm getting failures that seem decipherable.  In the "make lint" task:
```
stratis_cli_cert.py:144: [E0012(bad-option-value), ] Bad option value 'bad-continuation'
```
That is known, because that version of pylint (2.6.0-2) removed the option "bad-continuation".

**Comment by mulkieran** [comment]
> Now I'm getting failures that seem decipherable. In the "make lint" task:
> 
> ```
> stratis_cli_cert.py:144: [E0012(bad-option-value), ] Bad option value 'bad-continuation'
> ```
> 
> That is known, because that version of pylint (2.6.0-2) removed the option "bad-continuation".

Ok! We would be happy to see such failures in nightly, and we do, which is good. This is our regular CI, though, so we don't want to see any failures. And we want to be running test on fedora:32 containers, because that's what we are all currently using for development.

**Comment by bgurney-rh** [comment]
OK, I don't know if this is what you intended, but I'm trying a run with Fedora 32.  Let's see how that goes.

**Comment by bgurney-rh** [comment]
There, I think I got it.  Closing this to replace with a final PR.

**Review by mulkieran — CHANGES_REQUESTED** [review]
Ok. Once we drop python3-dbus we can add back in the test for LOWEST SUPPORTED INTERPRETER, which got dropped: https://github.com/stratis-storage/testing/pull/30/files#diff-7829468e86c1cc5d5133195b5cb48e1ff6c75e3e9203777f6b2e379d9e4882b3L31. We want to run it on fedora:29 as a proxy for the dependency numbers we really want to check.

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -19,36 +19,31 @@ jobs:
...
-          - python-version: 3.8.6
-            dependencies: >
-              dbus-python==1.2.16
-              pylint==2.4.4
-              psutil==5.6.7
+          - dependencies: >
+              dbus-python-devel
+              pylint
+              python3-psutil
+              python3-dbus
```
[the remark this answers is not in the document]
Comment (bgurney-rh) [reply .github/workflows/main.yml:26]: Without that line, I got an error of "unable to import 'dbus'."  So how do I satisfy that dependency?

**Comment by mulkieran** [comment]
@bgurney-rh I see your problem...

**Comment by bgurney-rh** [comment]
Yeah; the python3-dbus package in F32 provides `/usr/lib64/python3.8/site-packages/dbus` and its contents, while the dbus-python-devel package doesn't provide much, beyond `/usr/lib64/pkgconfig/dbus-python.pc` and `/usr/include/dbus-1.0/dbus/dbus-python.h`, and some documents.

**Pull request closed without merging**